### DISCUSSION TITLE
Is it possible to change log levels "on the fly"?

### DISCUSSION THREAD

**Discussion by LB-personal** [gabime/spdlog#2196] [Q&A]
If log level is given by environment variable, and the value of the variable change, will the log level of the logger change as well? and if not, is calling `spdlog::cfg::load_env_levels()` more then once and after loggers where created is supported?

**Comment by tt4g** [comment]
Changes in the environment variable are not monitored, so updating the environment variable will not change the logger's log level.

> `spdlog::cfg::load_env_levels()` more then once and after loggers where created is supported?

You can do it.

Note that environment variables are managed on a per-process basis.
Changing the system's environment variables does not usually update the process's environment variables.
You will need to update the environment variables in the process.
